File: go-client-sdk/pkg/models/operations/getonecrmemail.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmEmailQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmEmailResponseBody struct {
	Email *shared.Email `json:"email,omitempty"`
}

type GetOneCrmEmailRequest struct {
	QueryParams GetOneCrmEmailQueryParams
}

type GetOneCrmEmailResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmEmailResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmemail.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmEmailQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmEmailResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email?: shared.Email;
}


export class GetOneCrmEmailRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmEmailQueryParams;
}


export class GetOneCrmEmailResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmEmailResponseBody;
}
